tests/ui/rfcs/rfc-2632-const-trait-impl/tilde-const-assoc-fn-in-trait-impl.rs

```diff
@@ -0,0 +1,29 @@
+// Regression test for issue #119700.
+// check-pass
+
+#![feature(const_trait_impl, effects)]
+
+#[const_trait]
+trait Main {
+    fn compute<T: ~const Aux>() -> u32;
+}
+
+impl const Main for () {
+    fn compute<T: ~const Aux>() -> u32 {
+        T::generate()
```

Comment:
> Wait, the UI test is not a true reproducer for the linked issue, oops!

Well, it would be a reproducer of the ICE if I replaced the body with `0`

```suggestion
        0
```

but then the test would become a bit *boring* since we wouldn't actually use anything from `~const Aux`. With `T::generate()` nightly rustc incorrectly fails with `` mismatched types // expected `host`, found `true` `` since the ICE is a delayed bug. Ofc I could add revisions to this test or another file.

Reply:
It's not that big of a deal. Showing that it doesn't ICE (we can lower the where clause) and that it works correctly seems like a fine test.